tv_apps/tv_shared/js/card_manager.js

```diff
@@ -553,8 +553,13 @@
         } else if (query.manifestURL && card.nativeApp &&
             card.nativeApp.manifestURL === query.manifestURL) {
 
-          found = card;
-          return true;
+          // The launchURL only happens to AppBookmark, we only need to check if
+          // launchURL is exactly the same in this case.
+          if (!(card instanceof AppBookmark) ||
```

Comment:
Yes, you are correct... We need to check the launchURL only when the card is AppBookmark.